src/main/java/tahub/contacts/storage/JsonAdaptedGrade.java

```diff
@@ -0,0 +1,60 @@
+package tahub.contacts.storage;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import tahub.contacts.commons.exceptions.IllegalValueException;
+
+/**
+ * Jackson-friendly version of a grade entry.
+ */
+class JsonAdaptedGrade {
+
+    private final String assessmentName;
+    private final double score;
+    private final double weight;
+
+    /**
+     * Constructs a {@code JsonAdaptedGrade} with the given grade details.
+     */
+    @JsonCreator
+    public JsonAdaptedGrade(@JsonProperty("assessmentName") String assessmentName,
+                            @JsonProperty("score") double score,
+                            @JsonProperty("weight") double weight) {
+        this.assessmentName = assessmentName;
+        this.score = score;
+        this.weight = weight;
+    }
+
+    public String getAssessmentName() {
+        return assessmentName;
+    }
+
+    public double getScore() {
+        return score;
+    }
+
+    public double getWeight() {
+        return weight;
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted grade object into the model's grade data.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted grade.
+     */
+    public void toModelType(tahub.contacts.model.grade.GradingSystem gradingSystem) throws IllegalValueException {
```

Comment:
Should the ```toModelType``` method in the adapter deserialise the this ```JsonAdaptedGrade``` to a ```Grade``` java object? I think you are missing the ```Grade``` class if you want to implement ```GradingSystem``` as a list of ```Grades```. I might be mistaken though, lets wait for other comments :)